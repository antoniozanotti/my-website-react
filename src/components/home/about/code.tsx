import { useTranslations } from "next-intl";

export default function Code() {
  const tAbout = useTranslations('About');
  const t = useTranslations('');
  const colorConst = "text-accent-2 dark:text-dark-accent-2";
  const colorIdentifier = "text-accent-1 dark:text-dark-accent-1";
  const colorObject = "text-accent dark:text-dark-accent";
  const colorString = "text-primary-2 dark:text-dark-primary-2";
  return (
    <code>
      <span className={colorConst}>const</span>{" "}
      <span className={colorIdentifier}>{tAbout('varBirthday')}</span> ={" "}
      <span className={colorConst}>new</span>{" "}
      <span className={colorObject}>Date</span>(
      <span className={colorString}>{"\"1990-04-18\""}</span>);
      <br />
      <span className={colorConst}>const</span>{" "}
      <span className={colorIdentifier}>{tAbout('varLocation')}</span> ={" "}
      <span className={colorString}>{"\"Florian√≥polis, "+t("Brazil")+"\""}</span>;<br />
      <span className={colorConst}>const</span>{" "}
      <span className={colorIdentifier}>{tAbout('varPronouns')}</span> = [
      <span className={colorString}>{"\""+tAbout("pronouns.he")+"\""}</span>,{" "}
      <span className={colorString}>{"\""+tAbout("pronouns.him")+"\""}</span>];
      <br />
      <span className={colorConst}>const</span>{" "}
      <span className={colorIdentifier}>{tAbout('varDegree')}</span> ={" "}
      <span className={colorString}>
        {"\""+tAbout("degree")+"\""}
      </span>
      ;<br />
      <span className={colorConst}>const</span>{" "}
      <span className={colorIdentifier}>{tAbout('varValues')}</span> = [
      <span className={colorString}>{"\""+tAbout("values.ethic")+"\""}</span>,{" "}
      <span className={colorString}>{"\""+tAbout("values.love")+"\""}</span>,{" "}
      <span className={colorString}>{"\""+tAbout("values.freedom")+"\""}</span>];
    </code>
  );
}
