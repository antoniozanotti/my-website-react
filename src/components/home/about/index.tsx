import Section from "@/components/ui/section";
import Grid from "@/components/ui/grid";
import Code from "./code";

export default function Index() {
  return (
    <Section title="About" anchor="about">
      <Grid>
        <div className="flex flex-col space-y-4 mt-4">
          <p>
            I was born in Gramado, a small tourist city in the south of Brazil,
            today I live in the city of Florianópolis. I started my journey when
            I was 16 years old, in 2006. I have 18 years of experience in
            different types of projects: multinational ecommerce platform,
            social recipe network, real estate websites, hotel network…
          </p>

          <p>
            I have experience with Vue, React, Next.js, Vite, TypeScript,
            TanStack Query, Axios, Storybook, UnoCSS + Tailwind CSS, Redux,
            Pinia, GraphQL, Vitest, Jest, Playwright, which are the most used
            technologies currently for frontend. I also have experience with
            PHP, Magento, CakePHP, Zend Framework, Data Modeling, SQL, MySQL,
            PostgreSQL, PL/pgSQL. I’m very experienced with Figma, HTML, CSS and
            JavaScript.
          </p>

          <p>
            For most of my career, I worked as a Full Stack PHP developer. From
            2023 onwards, I returned to the frontend, as it is the area of
            software development that I most identify with. I’m not a ready
            professional, and I never will be, because I’ll always be learning
            something more. I don’t feel comfortable in a leadership position
            today, but it’s something I want to develop.
          </p>

          <p>
            Besides work, I love my husband, we’ve been living together for 7
            years. I also like to study sociology, do exercises at home, take a
            track, go to the beach, hangout with friends, watercolour,
            arduino...
          </p>
        </div>
        <div className="mt-4">
          <Code />
        </div>
      </Grid>
    </Section>
  );
}
