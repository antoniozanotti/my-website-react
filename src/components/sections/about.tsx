import Section from "@/components/ui/section";
import Grid from "@/components/ui/grid";
import Code from "@/components/blocks/code";

export default function About() {
  return (
    <Section title="# About" anchor="about">
      <Grid>
        <div className="flex flex-col space-y-4">
          <p>
            Senior Full Stack Developer with more than 15 years of experience.
            Looking for long-term opportunities to work as Frontend Developer.
          </p>
          <p>
            Comfortable with Vue or React, Next.js or Vite, Storybook,
            UnoCSS+Tailwind CSS, TypeScript, GraphQL… I’m open mind to study and
            adapt me to your Tech Stack.
          </p>
          <p>
            I’m friendly coworker, easy going, observant, curious and I like to help and listen. My place is between backend developers and designers.
          </p>
          <p>
            Besides work, I love my husband, we’ve been living together for 7
            years. I also like to study sociology, do exercises at home, take a
            track, go to the beach, hangout with friends, watercolour, arduino...
          </p>
        </div>
        <div className="mt-4 sm:mt-0">
          <Code />
        </div>
      </Grid>
    </Section>
  );
}
