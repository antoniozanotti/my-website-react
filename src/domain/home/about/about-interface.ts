export interface AboutInterface {
  paragraphs: string[]
}