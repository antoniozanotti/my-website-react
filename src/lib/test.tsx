import React, { ReactElement } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";
import { Theme } from "@/domain/theme/theme/theme";
import { Providers } from "../app/providers";
import messages from "@/messages/en.json";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return (
    <Providers locale="en" messages={messages}>
      <Theme>
        <div data-testid="container">{children}</div>
      </Theme>
    </Providers>
  );
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
  };
});
