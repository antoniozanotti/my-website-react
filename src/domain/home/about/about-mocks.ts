import { AboutInterface } from "@/domain/home/about/about-interface";

export const AboutMocks: AboutInterface = {
  paragraphs: ["About1", "About2"],
};
