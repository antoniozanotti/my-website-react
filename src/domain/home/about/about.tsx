import { AboutInterface } from "./about-interface";
import { useTranslations } from "next-intl";
import { Section } from "@/domain/ui/section/section";
import { Grid } from "@/domain/ui/grid/grid";
import { Code } from "../code/code";

export function About({ paragraphs }: AboutInterface) {
  const t = useTranslations("About");

  return (
    <Section title={t("title")} anchor="about">
      <Grid>
        <div className="flex flex-col space-y-4 mt-4">
          {paragraphs.map((p: string, index: number) => (
            <p key={index}>{p}</p>
          ))}
        </div>
        <div className="mt-4">
          <Code />
        </div>
      </Grid>
    </Section>
  );
}
