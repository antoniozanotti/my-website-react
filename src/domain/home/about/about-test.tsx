import { test, expect } from '@playwright/experimental-ct-react';
import { About } from "./about";
import { AboutMocks } from './about-mocks';

test.describe("About", () => {
  test("should render paragraphs", async ({ mount }) => {
    const component = await mount(<About paragraphs={AboutMocks.paragraphs} />);
    await expect(component).toContainText(AboutMocks.paragraphs[0]);
    await expect(component).toContainText(AboutMocks.paragraphs[1]);
  });
});
