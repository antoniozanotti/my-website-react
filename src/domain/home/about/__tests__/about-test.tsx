import { render, screen } from "@/lib/test";
import { About } from "../about";
import { AboutMocks } from "./about-mocks";

describe("About", () => {
  it("should render paragraphs", async () => {
    const { container } = render(<About paragraphs={AboutMocks.paragraphs} />);
    const paragraph1 = screen.getByText(AboutMocks.paragraphs[0]);
    const paragraph2 = screen.getByText(AboutMocks.paragraphs[1]);

    expect(paragraph1).toBeVisible();
    expect(paragraph2).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
