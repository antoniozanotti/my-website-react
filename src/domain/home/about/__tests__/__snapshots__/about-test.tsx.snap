// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should render paragraphs 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <section>
          <h2
            class="leading-none text-c9 sm:text-c10 lg:text-c11 text-accent-1 dark:text-dark-accent-1 drop-shadow mt-[164px] "
            id="about"
          >
            About
          </h2>
          <div
            class="sm:grid sm:gap-x-8 lg:gap-x-16 sm:grid-cols-2"
          >
            <div
              class="flex flex-col space-y-4 mt-4"
            >
              <p>
                About1
              </p>
              <p>
                About2
              </p>
            </div>
            <div
              class="mt-4"
            >
              <code>
                <span
                  class="text-accent-1 dark:text-dark-accent-1"
                >
                  const
                </span>
                 
                <span
                  class="text-accent-1 dark:text-dark-accent-1"
                >
                  birthday
                </span>
                 =
                 
                <span
                  class="text-accent-1 dark:text-dark-accent-1"
                >
                  new
                </span>
                 
                <span
                  class="text-accent dark:text-dark-accent"
                >
                  Date
                </span>
                (
                <span
                  class="text-primary-2 dark:text-dark-primary-2"
                >
                  "1990-04-18"
                </span>
                );
                <br />
                <span
                  class="text-accent-1 dark:text-dark-accent-1"
                >
                  const
                </span>
                 
                <span
                  class="text-accent-1 dark:text-dark-accent-1"
                >
                  location
                </span>
                 =
                 
                <span
                  class="text-primary-2 dark:text-dark-primary-2"
                >
                  "Florian√≥polis, Brazil"
                </span>
                ;
                <br />
                <span
                  class="text-accent-1 dark:text-dark-accent-1"
                >
                  const
                </span>
                 
                <span
                  class="text-accent-1 dark:text-dark-accent-1"
                >
                  pronouns
                </span>
                 = [
                <span
                  class="text-primary-2 dark:text-dark-primary-2"
                >
                  "he"
                </span>
                ,
                 
                <span
                  class="text-primary-2 dark:text-dark-primary-2"
                >
                  "him"
                </span>
                ];
                <br />
                <span
                  class="text-accent-1 dark:text-dark-accent-1"
                >
                  const
                </span>
                 
                <span
                  class="text-accent-1 dark:text-dark-accent-1"
                >
                  degree
                </span>
                 =
                 
                <span
                  class="text-primary-2 dark:text-dark-primary-2"
                >
                  "Internet Systems Technologist (FEEVALE, 2010)"
                </span>
                ;
                <br />
                <span
                  class="text-accent-1 dark:text-dark-accent-1"
                >
                  const
                </span>
                 
                <span
                  class="text-accent-1 dark:text-dark-accent-1"
                >
                  values
                </span>
                 = [
                <span
                  class="text-primary-2 dark:text-dark-primary-2"
                >
                  "ethic"
                </span>
                ,
                 
                <span
                  class="text-primary-2 dark:text-dark-primary-2"
                >
                  "love"
                </span>
                ,
                 
                <span
                  class="text-primary-2 dark:text-dark-primary-2"
                >
                  "freedom"
                </span>
                ];
              </code>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;
